src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,368 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/support/CommonIterator.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+inline constexpr size_t kLocationNameMaxSize = 128u;
+inline constexpr size_t kMapNameMaxSize      = 64u;
+
+/**
+ * This class is used to wrap the LocationStruct object and provide a more user-friendly interface for the data.
+ * It provides a way to store the location name in a buffer, and provides a way to compare the location name with a given string.
+ */
+struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::LocationStruct::Type
+{
+    /**
+     * @brief This is a default constructor that initializes the location object with default values.
+     */
+    LocationStructureWrapper()
+    {
+        Set(0, 0, CharSpan(), DataModel::Nullable<int16_t>(), DataModel::Nullable<AreaTypeTag>(),
+            DataModel::Nullable<LandmarkTag>(), DataModel::Nullable<PositionTag>(), DataModel::Nullable<FloorSurfaceTag>());
+    }
+
+    /**
+     * @brief This is a full constructor that initializes the location object with the given values. All values are deep copied.
+     * @param[in] aLocationId The unique identifier of this location.
+     * @param[in] aMapId The identifier of the supported map associated with this location.
+     * @param[in] aLocationName A human readable name for this location (empty string if not used).
+     * @param[in] aFloorNumber The floor level of this location - use negative values for below ground.
+     * @param[in] aAreaTypeTag A common namespace Area tag - indicates an association of the location with an indoor or outdoor area
+     * of a home.
+     * @param[in] aLandmarkTag A common namespace Landmark tag - indicates an association of the location with a home landmark.
+     * @param[in] aPositionTag A common namespace Position tag - indicates the position of the location with respect to the
+     * landmark.
+     * @param[in] aSurfaceTag A common namespace Floor Surface tag - indicates an association of the location with a surface type.
+     *
+     * @note Requirements regarding what combinations of fields and values are valid are not checked by this class.
+     * @note If aLocationName is larger than kLocationNameMaxSize, it will be truncated.
+     * @note If aLocationName is an empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
+     */
+    LocationStructureWrapper(uint32_t aLocationId, const DataModel::Nullable<uint8_t> & aMapId, const CharSpan & aLocationName,
+                             const DataModel::Nullable<int16_t> & aFloorNumber,
+                             const DataModel::Nullable<AreaTypeTag> & aAreaTypeTag,
```

Comment:
nit: I would prefer not to use the `a*` for hungarian-style argument naming. E.g. STL does not do that either.

Reply:
Do you prefer these not to have the `a` in front? This seems to be quite often used in the SDK and I seem to remember having had reviewers arguing in favour of it.

Reply:
I am arguing against having a `a` prefix for new files. We inherited some files from weave using that style and there I can undestand and argument of `be consistent with the file`. However for new files I find the practice frustrating.

Reply:
This is tracked by [this issue](https://github.com/project-chip/connectedhomeip/issues/34541).